File: implementations/csharp/src/Loom.Core/Models/Message.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    public class Message
    {
        public Enums.MessageRole Role { get; set; }
        public string Content { get; set; }
        public string ToolCallId { get; set; }
        public string ToolName { get; set; } 
        public Dictionary<string, object> ToolArgs { get; set; }
        public Dictionary<string, object> Metadata { get; set; }
    }
}

File: implementations/vb/Loom.Core/Models/Message.vb
Namespace Models
    ''' <summary>
    ''' Single message
    ''' </summary>
    Public Class Message
        Public Property Role As Enums.MessageRole
        Public Property Content As String
        Public Property ToolCallId As String
        Public Property ToolName As String
        Public Property ToolArgs As New Dictionary(Of String, Object)
        Public Property Metadata As New Dictionary(Of String, Object)
    End Class
End Namespace